### PR TITLE
[rv_dm] Declare RV_DM debug memory as RW region

### PR DESCRIPTION
**Pull request by msfschaffner**
Follow up for #15166.

Since the RV_DM debug memory contains both writeable and readable regions, the corresponding window in `rv_dm.hjson` needs to be declared as an RW region (otherwise we may see spurious RAL model errors in the top-level simulation). This PR also renames the TL-UL port from ROM to MEM, since ROM could be misleading.

This should address https://github.com/lowRISC/opentitan/issues/14200, https://github.com/lowRISC/opentitan/issues/14653 and https://github.com/lowRISC/opentitan/issues/14921.

(Note that the window in `rv_dm.hjson` is only a stub. We model the memory and registers inside the vendored in debug module more accurately in `rv_dm_debug_mem.hjson`.)

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

**Review by jdonjdon — APPROVED** [review]
LGTM

File: hw/ip/rv_dm/data/rv_dm.hjson [hunk 1]
```diff
@@ -102,13 +102,14 @@
...
+    mem: [
+      // Debug memory size (given as `items` below) must be a power of two.
       { window: {
-          name: "ROM"
+          name: "MEM"
           items: "1024" // 4 KiB
-          swaccess: "ro",
-          desc: '''Access window into the debug ROM.'''
+          swaccess: "rw",
+          byte-write: "true",
```
Comment (msfschaffner) [inline hw/ip/rv_dm/data/rv_dm.hjson:111]: this is the relevant part of the change. the remaining diffs are only due to the name change.

Comment (tjaychen) [reply hw/ip/rv_dm/data/rv_dm.hjson:111]: was the previous attribute actually doing anything inside `rv_dm`..? i can't remember. Or was this more for the testbench?

Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:111]: No, the RTL was not affected, but the RAL model uses this for predictions.

Comment (tjaychen) [reply hw/ip/rv_dm/data/rv_dm.hjson:111]: sounds good.

**Review by tjaychen — APPROVED** [review]

**Comment by sriyerg** [comment]
This will likely break rv dm and chip level mem access tests.

**Comment by msfschaffner** [comment]
> This will likely break rv dm and chip level mem access tests.

Do you have any specific tests in mind that we should run?

I ran both the `rv_dm` block level tests and the `chip_csr_mem_rw_with_rand_reset` and `chip_tl_errors` top-level tests with multiple seeds:
```
- util/dvsim/dvsim.py hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson --reseed 50
- util/dvsim/dvsim.py hw/top_earlgrey/dv/chip_sim_cfg.hjson -i chip_csr_mem_rw_with_rand_reset chip_tl_errors --reseed 50
```
The block-level tests and `chip_csr_mem_rw_with_rand_reset` test passed 100%.
There were still some failures in the `chip_tl_errors` test, but they seem unrelated since the failure addresses are not hitting the `rv_dm` memory space. The patch increases the pass rate of `chip_tl_errors` from ~33% to ~92%.

**Comment by msfschaffner** [comment]
Last force push fixes some CI errors in the SW build (I overlooked a few software constants that had to be renamed).

**Comment by msfschaffner** [comment]
I am merging this so that the nightlies can pick this up. I will check the results tomorrow to see whether there is any fallout that needs to be fixed.

**Comment by vogelpi** [comment]
Many thanks @msfschaffner for digging into this and fixing it, this looks great!

**Pull request merged**